lumicks/pylake/detail/confocal.py

```diff
@@ -8,6 +8,25 @@
 from .image import reconstruct_image_sum, reconstruct_image, save_tiff, ImageMetadata
 
 
+def _default_image_factory(self, channel):
+    channel_data = getattr(self, f"{channel}_photon_count").data
+    raw_image = reconstruct_image_sum(channel_data, self.infowave.data, self._shape)
+    return self._to_spatial(raw_image)
```

Comment:
I know that I originally factored this out, but looking at it now, it would be good to add a type annotation so that we know what this `self` is intended to be:
```py
def _default_image_factory(self: "ConfocalImage", channel):
```
Same thing for the other factories.

Reply:
Good point.